### ISSUE TITLE
Does 2.5.8 result in widespread failure of common dataviz?

### ISSUE THREAD

**Issue by scottaohara** [w3c/wcag#3091]
reading through https://github.com/w3c/wcag/pull/3089, it dawned upon me that this rule has no mention of, or exception for dataviz which from my understanding would often fail this rule.

For instance, a [scatter plot chart](https://www.highcharts.com/demo/scatter) where often data points overlap each other, and in more elaborate examples than the one I posted, each data point can link to more information about that particular point.

Even github's yearly data charts on each user's profile are presently 10x10 squares with a few pixels of space between them.  That would fail this rule too?

**Comment by patrickhlauke** [comment]
I guess this will be a case of having to pass it under the "Essential" exception

> Essential: A particular presentation of the target is [essential](https://w3c.github.io/wcag/guidelines/22/#dfn-essential) or is legally required for the information being conveyed;

**Comment by scottaohara** [comment]
yeh, i was thinking about the "essential" exception and kept going and forth on whether or not it applied.  how essential is any specific dataviz design?  just seems to me like it'd be far better to call stuff like this out directly than allow for more opportunities for people to interpret things differently.

**Comment by patrickhlauke** [comment]
with "Essential" clauses, my first instinct is always to ask "Essential to *whom*?" ...

**Comment by JAWS-test** [comment]
https://w3c.github.io/wcag/understanding/target-size-minimum.html

> For example, in digital maps, the position of pins is analogous to the position of places shown on the map. If there are many pins close together, the offset between pins and neighboring pins will often be below 24 CSS pixels. It is essential to show the pins at the correct map location, therefore the Essential exception applies. 

What is described here for maps, applies in my opinion analogously for charts

**Comment by JAWS-test** [comment]
@patrickhlauke 

> with "Essential" clauses, my first instinct is always to ask "Essential to whom?" ...

essential for the information or functionality

"essential: if removed, would fundamentally change the information or functionality of the content, and information and functionality cannot be achieved in another way that would conform"

https://www.w3.org/TR/WCAG21/#dfn-essential

**Comment by patrickhlauke** [comment]
yeah, thanks @JAWS-test, I'm aware

**Comment by scottaohara** [comment]
thanks @JAWS-test.  I suppose one could make the comparison between a map and dataviz.  I didn't initially.  I can see your point.  I'd still rather that be stated though if that's actually the intent.

>I can also design the charts and maps in such a way that the target has sufficient distance...

right, but many people don't design their own charts and maps.  so what would they do?

Re: if one would be expected to use zoom in / out to make the target size/spacing larger for a map coordinate/dataviz data point, why couldn't that be true for any UI?  Why should there be a normative rule to make the button larger / have more space if someone can just zoom in their browser and effectively bump it up to be larger than 24x24.    That would seem to gut the point of this SC, imo.

**Comment by JAWS-test** [comment]
@scottaohara 

> Re: if one would be expected to use zoom in / out to make the target size/spacing larger for a map coordinate/dataviz data point, why couldn't that be true for any UI?

The point of 2.5.8 and 2.5.5 is that without zoom, the targets are large enough. But there is the essential exception. And if I use this, it would be good (as a recommendation in Understanding) to allow zoom. This is sometimes not the case: while I can scale the rest of the page without any problems, I can't always do it with maps and charts, because they adapt to the viewport width.

**Comment by scottaohara** [comment]
ok, thanks for clarifying what you meant (the map/viz adapting to the viewport).

I get the point of the rule is without zoom, hence my bringing up that mentioning one can zoom would be against the purpose of the UI.  But your example, if clarified that it was allowed in instances where the map/dataviz automatically adapted to the viewport, and thus _needed_ its own zooming mechanism, then that'd make sense.  Without that clarity, I'd just see the mention of zoom as a red flag for people to be like "well, this all seems rather arbitrary. why not just always allow zoom?"

**Comment by aardrian** [comment]
I appreciate that @JAWS-test is offering some other insights and even possible workarounds, but I fear that could let the Working Group folks de-prioritize this discussion when clarity is needed from _them_.

To reinforce the original issue, regardless of the follow-on conversations, I would also like clarity regarding data visualizations under 2.5.8 from the WG. Ideally in the form of documentation.

**Comment by mbgower** [comment]
So, to clarify the discussion, we are talking about a data visualization where every data point in a scatter plot (or similar chart) is a UIC component, correct?
I'm going to pause for a moment to note that I'm not sure how common those are. High charts has pulled it off, but I'm not sure how many other fully interactive, accessible complex data visualizations exist in the world -- or how useful a chart would be that had tiny data points all over the place -- or how fun (or necessary) it would be for any user to click on a 2px target.

However, in the hypothetical where a scatter plot graph strews tiny interactive data points all over the place (based on real-time data, for example), it would fall under essential for exactly the same reasons the map would. How else could it be shown?
I would also expect such a graph -- if the designer wanted users to maintain their equilibrium -- to provide zoom functions similar to a map so that the user could interact with the data at a reasonable scale.

Does anyone think interactive data visualizations should _not_ fall under the exception?

**Comment by bruce-usab** [comment]
I take a stab, and tie Essential to information density and parallel to sensory.  I will also emphasis Equivalent exception over Essential.

**Comment by bruce-usab** [comment]
@rachaelbradley and @alastc -- Friday I added to Understanding which I believe already addresses the concerns from this thread.
https://github.com/w3c/wcag/blob/understanding-focus-not-obscured/understanding/22/focus-not-obscured-minimum.html

But I did _not_ create that branch and was pleasantly surprised how complete it looked to me.  I was going to ask on today's backlog call.

**Comment by bruce-usab** [comment]
My apologies @rachaelbradley I am mixing up work needed on Understanding for obscuring focus versus target size.  I have _not_ started on the later.  I do think there is phrasing (wrt essential) from that branch which may of utility.

**Comment by bruce-usab** [comment]
See:  https://github.com/w3c/wcag/pull/3188/files for my prosed changes to Understanding.

Some editorial, but task from this issues thread was adding:
> A similar example is an interactive data visualization where targets are necessarily dense. ... When the essential exception is applicable, authors are strongly encouraged to provide equivalent functionality through alternative means to the extent practical.

I also added a paragraph of examples under Intent:

> Disabilities addressed by this requirement include hand tremors, spasticity, and quadriplegia.  Some people with disabilities use specialized input devices instead of a computer mouse or trackpad.  Typically these types of input device do not provide as much accuracy as mainstream pointing devices.  Meeting this requirement also ensures that touchscreen interfaces are easier to use.

**Comment by mbgower** [comment]
> For data visualization, providing the data used to create the visualization is the best practice and provides an accessible alternative to a data visualization which is an enhancement

I don't think this relates to the discussion on target size in this issue. The specific scenario here is that the data visualization is made up of data points that are targets. In such a situation, I think a tabular display of static text is the (inadequate) alternative.

**Comment by jenstrickland** [comment]
> > For data visualization, providing the data used to create the visualization is the best practice and provides an accessible alternative to a data visualization which is an enhancement
> 
> 
> 
> I don't think this aligns to the discussion on target size in this issue. The specific scenario here is that the data visualization is made up of data points that are targets. In such a situation, I think a tabular display of static text is the (inadequate) alternative.


I think it does relate, as it is information that could be presented in an understanding document. For situations where a data viz may be designed, a designer would be expected to consider meeting the criteria. If they can't, which is likely for a variety of reasons, they could provide access to the tabular data. 

My comment was sort of a sidebar -- as I voted in favor of the update in the survey. The comment was provided for use as the understanding content is developed.

**Comment by mbgower** [comment]
> For situations where a data viz may be designed, a designer would be expected to consider meeting the criteria. If they can't, which is likely for a variety of reasons, they could provide access to the tabular data.

I'm not disputing that providing tabular data as a fall back for charts is a good strategy. Carbon Charts has [included that as a basic feature](https://medium.com/carbondesign/color-palettes-and-accessibility-features-for-data-visualization-7869f4874fca) for many years. I'm just saying I don't see how it is relevant to target size.

**Comment by junykallukalam** [comment]
Your concern about data visualization (dataviz) elements, like scatter plots or GitHub's yearly data charts, potentially failing this rule is valid. If the data charts are essential, GitHub could provide alternative ways to access the same information for users who might struggle with the small target sizes, such as a tabular data view or additional navigation options.
